### PR TITLE
Support static valued scalar arrays in expand_dims/squeeze

### LINKED ISSUE #3243: jax.numpy argument handling does not work for DeviceArray in lieu of int
I just ran into a bug when trying to import into google3 the changes from #3217. One internal test failed because it passes `DeviceArray(-1)` for `axis` in a call to `jax.numpy.expand_dims`. This used to work before #3217. 

The problem is that `lax.numpy.expand_dims` (and many other functions) use code like 
```
if isinstance(axis, int):
    axis = (axis,)
```

This code fails if we pass a DeviceArray(-1) (the failure is [here](https://github.com/google/jax/blob/master/jax/interpreters/xla.py#L943))

I can see how the user may end up with a DeviceArray for that argument, e.g., when they used some on-device computation for it. 

I think we need to handle this systematically. For now, I am going to just make a small fix for `expand_dims` only.

### PR DESCRIPTION
**Pull request by shoyer**
Fixes https://github.com/google/jax/issues/3243

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Excellent!

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -1145,19 +1145,16 @@ def squeeze(a, axis: Union[int, Tuple[int, ...]] = None):
...
 
 @_wraps(np.expand_dims)
 def expand_dims(a, axis: Union[int, Tuple[int, ...]]):
-  # TODO(https://github.com/google/jax/issues/3243)
-  try:
-    axis = (int(axis),)  # type: ignore[arg-type]
-  except TypeError:
-    pass
-  return lax.expand_dims(a, axis)  # type: ignore[arg-type]
+  if not isinstance(axis, tuple):
```
Comment (gnecula) [inline jax/numpy/lax_numpy.py:1155]: I am wondering how is this code going to behave if somebody passes a DeviceArray for axis? Specifically, I am wondering if the rollback that I hit on Friday will hit again. 

In general, for changes that may break clients, even abusing ones, we should be aware that the breakage will be manifested through an import failure. It would be best to try such a PR inside google3 (go/jax-source-sync) or at least to alert the oncall.

Comment (shoyer) [reply jax/numpy/lax_numpy.py:1155]: I was lazy here with my tests. I used a scalar NumPy array instead of a DeviceArray because you can’t put DeviceArray objects in decorators. But they work the same in this context (converted via operator.index)
